### DISCUSSION TITLE
DBLE date分区类型使用 >、< 查询时间范围会下发到所有分区？

### DISCUSSION THREAD

**Discussion by wangjuan-action** [actiontech/dble#3425] [Q&A] [answered]
配置date类型分区，配置可参考issue：https://github.com/actiontech/dble/issues/3423

使用between... and...语句只查询对应时间范围内的分区，使用<、>语句时却下发到所有分区

> mysql> explain select * from date_test where log_date >= '2022-01-29 00:00:00' and log_date <= '2022-03-29 00:00:00';
> +---------------+----------+------------------------------------------------------------------------------------------------------------------+
> | SHARDING_NODE | TYPE     | SQL/REF                                                                                                          |
> +---------------+----------+------------------------------------------------------------------------------------------------------------------+
> | dn1           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> | dn2           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> | dn3           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> | dn4           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> | dn5           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> | dn6           | BASE SQL | SELECT * FROM date_test WHERE log_date >= '2022-01-29 00:00:00'  AND log_date <= '2022-03-29 00:00:00' LIMIT 100 |
> +---------------+----------+------------------------------------------------------------------------------------------------------------------+
> 6 rows in set (0.00 sec)

> mysql> explain select * from date_test where log_date between '2022-01-29 00:00:00' and '2022-03-29 00:00:00';
> +---------------+----------+------------------------------------------------------------------------------------------------+
> | SHARDING_NODE | TYPE     | SQL/REF                                                                                        |
> +---------------+----------+------------------------------------------------------------------------------------------------+
> | dn1           | BASE SQL | select * from date_test where log_date between '2022-01-29 00:00:00' and '2022-03-29 00:00:00' |
> | dn2           | BASE SQL | select * from date_test where log_date between '2022-01-29 00:00:00' and '2022-03-29 00:00:00' |
> | dn3           | BASE SQL | select * from date_test where log_date between '2022-01-29 00:00:00' and '2022-03-29 00:00:00' |
> +---------------+----------+------------------------------------------------------------------------------------------------+
> 3 rows in set (0.01 sec)

**Comment by PanternBao** [comment, marked as answer]
between and 可以路由到特定节点，是因为有上界和下界。
\>和\< 等符号不支持是因为>和< 等符号可能是单边有界，所以为了处理方便，直接就广播下发。
